purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesOrchestrator.kt

```diff
@@ -110,6 +111,13 @@ internal class PurchasesOrchestrator(
             purchasesStateCache.purchasesState = value
         }
 
+    val latestConfiguration: PurchasesConfiguration
+        get() = if (initialConfiguration.appUserID == null) {
+            initialConfiguration
+        } else {
+            initialConfiguration.copy(appUserID = this.appUserID)
```

Comment:
We're now creating new copies every time this is read. Could we keep the up-to-date configuration in a backing `var` maybe? We can update it whenever the `appUserId` changes.

Reply:
Hmm I've been thinking whether this would be more work right now to calculate on user change or as it is now, when there is a configuration or the PaywallActivity's `onSaveInstanceState` is called... And I think it might be less work as it is right now? Normally users only configure once, so this won't be called, but there is a bigger chance of logging a different user (though also an uncommon case)... In any case I think we might be trying to early optimizing and I prefer to calculate it on the fly rather than trying to cache the value which might be outdated later... Lmk if you think otherwise!

Reply:
Yea this is less error prone, and the performance difference is probably negligible. I'm okay with it!